packages/api/src/providers/fitbit.ts

```diff
@@ -363,63 +370,50 @@ export class Fitbit extends Provider implements OAuth2 {
-      [FitbitScopes.sleep]: "sleep",
-      [FitbitScopes.weight]: "body",
-    };
-
-    const activeSubscriptions = await this.getActiveSubscriptions(accessToken);
-    if (activeSubscriptions)
-      await this.revokeTokenFromOtherUsers(user, fitbitUserId, activeSubscriptions, accessToken);
-
-    // TODO #652: Implement userRevokedAccess
+    await this.revokeTokenFromOtherUsers(user, fitbitUserId);
 
     // Creates new WH subscriptions based on the user's selected scopes
     await Promise.all(
-      Object.entries(subscriptionTypes).map(async ([key, subscriptionType]) => {
+      Object.entries(Fitbit.subscriptionTypes).map(async ([key, subscriptionType]) => {
         if (scopes.includes(key)) {
           const subscriptionId = `${fitbitUserId}-${subscriptionType}`;
           const subscriptionUrl = `${Fitbit.URL}/${Fitbit.API_PATH}/${subscriptionType}/apiSubscriptions/${subscriptionId}.json`;
-          this.createSubscription(subscriptionUrl, accessToken);
+          await this.createSubscription(subscriptionUrl, accessToken);
         }
       })
     );
   }
 
-  // Finds existing users connected to this Fitbit account, revokes their tokens and sends WH updates about the disconnections
-  async revokeTokenFromOtherUsers(
-    currentUser: ConnectedUser,
-    fitbitUserId: string,
-    activeSubscriptions: FitbitWebhookSubscriptions,
-    accessToken: string
-  ): Promise<void> {
+  /**
+   * Finds all users connected to the same Fitbit user, removes their tokens and deletes their WH subscriptions, if there are any.
+   *
+   * @param currentUser current ConnectedUser who is connecting to Fitbit
+   * @param fitbitUserId Fitbit ID
+   */
+  async revokeTokenFromOtherUsers(currentUser: ConnectedUser, fitbitUserId: string): Promise<void> {
     try {
-      if (activeSubscriptions.length > 0) {
-        const connectedUsers = await getConnectedUsersByTokenOrFail(
-          ProviderSource.fitbit,
-          fitbitUserId
-        );
-
-        await Promise.all(
-          connectedUsers.map(async user => {
-            if (
-              user.dataValues.id !== currentUser.dataValues.id &&
-              user.dataValues.cxId === currentUser.dataValues.cxId
-            ) {
-              await this.oauth.revokeLocal(user);
-              user = await getConnectedUserOrFail({
-                id: user.id,
-                cxId: user.cxId,
-              });
-              sendProviderDisconnected(user, [ProviderSource.fitbit]);
-            }
-          })
-        );
-        await this.deleteSubscriptions(activeSubscriptions, accessToken);
-      }
+      const connectedUsers = await getConnectedUsersByTokenOrFail(
+        ProviderSource.fitbit,
+        fitbitUserId
+      );
+
+      await Promise.all(
+        connectedUsers.map(async user => {
+          if (user.dataValues.id !== currentUser.dataValues.id) {
+            const token = getProviderTokenFromConnectedUserOrFail(user, ProviderSource.fitbit);
```

Comment:
Do we want to fail and error/interrupt the processing or can we fail gracefully, just skipping the processing of that `user` if there's no token for Fitbit?

Reply:
Changed this to allSettled, so one error won't stop the execution.
Also, in this case, there will always be a token for Fitbit because we're going thru the list of users found to have a token (just above) - it's just a matter of actually getting it the token itself.